### PR TITLE
tests: e2e test case for init with --skip-install flag

### PR DESCRIPTION
**Pull request by jamesgeorge007**
Summary:
---------
Follow up of #1143 

Test Plan:
----------
This PR adds a test case for the recently landed feature (CLI flag).

### CODE REVIEW

File: __e2e__/init.test.ts [hunk 1]
```diff
@@ -94,3 +94,32 @@ test('init --template file with custom directory', () => {
...
+  expect(stdout).toContain('Run instructions');
+
+  // make sure we don't leave garbage
+  expect(fs.readdirSync(DIR)).toContain('custom');
+
+  let dirFiles = fs.readdirSync(path.join(DIR, 'TestInit'));
+
+  expect(dirFiles).toEqual(
+    customTemplateCopiedFiles.filter(
+      file => !['node_modules', 'yarn.lock'].includes(file),
```
Comment (thymikee) [inline __e2e__/init.test.ts:122]: can we also check if those files exist in previous test "init --template file with custom directory"

Comment (jamesgeorge007) [reply __e2e__/init.test.ts:122]: We already have it 
https://github.com/react-native-community/cli/blob/2c08a8254cacae74ae2cbdd4e25d8506d4983d15/__e2e__/init.test.ts#L95

Comment (jamesgeorge007) [reply __e2e__/init.test.ts:122]: I assume you're looking forward to have another test case with the custom directory and `--skip-install` flag.

Comment (thymikee) [reply __e2e__/init.test.ts:122]: Nah, I don't care too much about it

Comment (jamesgeorge007) [reply __e2e__/init.test.ts:122]: I'm sorry, what do you expect more from my side in the context of this PR?

Comment (thymikee) [reply __e2e__/init.test.ts:122]: You're right, now I noticed `customTemplateCopiedFiles` includes these :)

**Pull request merged**